DEVEL.md

```diff
@@ -122,6 +122,20 @@ Otherwise, they can be logged as simple values:
     klog.Warningf("Example of a warning due to an external error: %v", err)
 ```
 
+### Checklist for new device plugins
+
+For new device plugins contributed to this repository, below is a
+checklist to get the plugin on par feature and quality wise with
+others:
+
+1. Plugin binary available in `cmd/`, its corresponding Dockerfile in `build/docker/` and deployment Kustomization/YAMLs in `deployments/`.
+2. Plugin binary Go unit tests implemented and passing with >80% coverage: `make test WHAT=./cmd/<device>`.
+3. Plugin binary linter chcks passing: `make lint`.
+4. Plugin e2e tests implemented and passing: `go test -v ./test/e2e/... -args -ginkgo.focus "<DEVICE>"`.
+5. Plugin CRD API added to `pkg/apis/deviceplugin/v1` and CRDs generated: `make generate`.
+6. Plugin CRD validation tests implemented in `test/envtest/` and passing: `make envtest`.
+7. Plugin CRD controller implemented in `pkg/controllers/` and added to the manager in `cmd/operator/main.go`.
```

Comment:
Should we have an item about corresponding documentation too?

Reply:
good comment, added